adr/ADR-42.md

```diff
@@ -0,0 +1,166 @@
+Acks, any other Ack policy will produce an error. If we force this ack policy in normal use we should error in Pedantic mode.
+
+Pull requests will have the following additional fields:
+
+ * `"group": "jobs"` - the group the pull belongs to, pulls not part of a valid group will result in an error
+ * `"min_pending": 1000` - only deliver messages when `num_pending` for the consumer is >= 1000
+ * `"min_ack_pending: 1000` - only deliver messages when `ack_pending` for the consumer is >= 1000
+
+If `min_pending` and `min_ack_pending` are both given either being satisfied will result in delivery (boolean OR).
+
+In the specific case where MaxAckPending is 1 and a pull is made using `min_pending: 1` this should only be served when
+there are no other pulls waiting. This means we have to give priority to pulls without conditions over those with when
+considering the next pull that will receive a message.
+
+### `pinned_client` policy
+
+Users want to have a single client perform all the work for a consumer, but they also want to have a stand-by client that
+can take over when the primary, aka `pinned` client, fails.
+
+**NOTE: We should not describe this in terms of exclusivity as there is no such guarantee, there will be times when one
+client think it is pinned when it is not because the server switched.** 
+
+The `pinned_client` policy provides server-side orchestration for the selection of the pinned client.
+
+```go
+{
+	PriorityGroups: ["jobs"],
+	PriorityPolicy: "pinned_client",
+	PriorityTimeout: 120*time.Second,
+	AckPolicy: "explicit",
+	// ... other consumer options
+}
+```
+
+This configuration states:
+
+ * We have 1 group defined and all pulls have to belong to this group
+ * The policy is `pinned_client` that activates these behaviors
+ * When a pinned client has not done any pulls in the last 120 seconds the server will switch to another client
+ * AckPolicy has to be `explicit`. If we force this ack policy in normal use we should error in Pedantic mode
+
+A pull request will have the following additional fields:
+
+ * `"group": "jobs"` - the group the pull belongs to, pulls not part of a valid group will result in an error
+ * `"id": "xyz"` - the pinned client will have this ID set to the one the server last supplied (see below), otherwise 
+   this field is absent
+
+After selecting a new pinned client, the first message that will be delivered to this client, and all future ones, will 
+include a Nats-Pin-Id: xyz header. The client that gets this message should at that point ensure that all future pull 
+requests have the same ID set.
+
+When a new pinned client needs to be picked - after timeout, admin action, first delivery etc, this process is followed:
+
+ 1. Stop delivering messages for this group, wait for all in-flight messages to be completed, continue to serve heartbeats
+ 2. Pick the new pinned client by rtt
```

Comment:
After call with Derek, we learned that the RTT we have access to in the server is not reliable and we should just pick the first available Request.

Reply:
AFAIK Stream Alternates are sorted by Gateways RTT, not client RTT.